AppDevPods/AppDevCameraKit/ADKCamera.m

```diff
@@ -0,0 +1,1342 @@
+
+        if (error) {
+            [self handleErrorWithCode:error.code userInfo:error.userInfo];
+        }
+    }
+}
+
+- (CGFloat)zoomFactor
+{
+    CGFloat zoomFactor;
+    if (self.videoCaptureDevice) {
+        zoomFactor = self.videoCaptureDevice.videoZoomFactor;
+    }
+
+    return zoomFactor;
+}
+
+- (void)setLensPosition:(CGFloat)lensPosition
+{
+    if (self.videoCaptureDevice) {
+        NSError *error;
+        if ([self.videoCaptureDevice lockForConfiguration:&error]) {
+            [self.videoCaptureDevice setFocusModeLockedWithLensPosition:lensPosition
+                                                      completionHandler:^(CMTime syncTime) {
+                                                          // Do nothing
+                                                      }];
+            [self.videoCaptureDevice unlockForConfiguration];
+        }
+
+        if (error) {
+            [self handleErrorWithCode:error.code userInfo:error.userInfo];
+        }
+    }
+}
+
+- (CGFloat)minLensPosition
+{
+    return 0.0f;
+}
+
+- (CGFloat)maxLensPosition
+{
+    return 1.0f;
+}
+
+- (CGFloat)lensPosition
+{
+    CGFloat lensPosition;
+    if (self.videoCaptureDevice) {
+        lensPosition = self.videoCaptureDevice.lensPosition;
+    }
+
+    return lensPosition;
+}
+
+- (void)setWhiteBalanceGains:(AVCaptureWhiteBalanceGains)whiteBalanceGains
+{
+    if (self.videoCaptureDevice) {
+        AVCaptureWhiteBalanceGains normalizedWhiteBalanceGains = [self normalizedGains:whiteBalanceGains];
+
+        NSError *error;
+        if ([self.videoCaptureDevice lockForConfiguration:&error]) {
+            [self.videoCaptureDevice setWhiteBalanceModeLockedWithDeviceWhiteBalanceGains:normalizedWhiteBalanceGains
+                                                                        completionHandler:^(CMTime syncTime) {
+                                                                            // Do nothing
+                                                                        }];
+            [self.videoCaptureDevice unlockForConfiguration];
+        }
+
+        if (error) {
+            [self handleErrorWithCode:error.code userInfo:error.userInfo];
+        }
+    }
+}
+
+- (AVCaptureWhiteBalanceGains)whiteBalanceGains
+{
+    AVCaptureWhiteBalanceGains whiteBalanceGains;
+    if (self.videoCaptureDevice) {
+        whiteBalanceGains = [self normalizedGains:[self.videoCaptureDevice deviceWhiteBalanceGains]];
+    }
+
+    return whiteBalanceGains;
+}
+
+- (CGFloat)minWhiteBalanceTemperature
+{
+    return 1700.0;
+}
+
+- (CGFloat)maxWhiteBalanceTemperature
+{
+    return 27000.0f;
+}
+
+- (void)setWhiteBalanceTemperature:(CGFloat)whiteBalanceTemperature
+{
+    // Color temperature definition of Kelvin https://en.wikipedia.org/wiki/Color_temperature#cite_note-4
+
+    if (self.videoCaptureDevice) {
+        AVCaptureWhiteBalanceTemperatureAndTintValues temperatureAndTintValues = {
+            .temperature = whiteBalanceTemperature,
```

Comment:
Just confirm, should we need compare with minWhiteBalanceTemperature and maxWhiteBalanceTemperature?